### PR TITLE
Use rlang dots helper instead of ellipsis

### PR DESCRIPTION
**Pull request by romainfrancois**
Perhaps this should drop the `rlang::` prefix too since `rlang` is `@import`'ed

### CODE REVIEW

File: R/rows.R [hunk 1]
```diff
@@ -186,7 +186,7 @@ rows_upsert.data.frame <- function(x, y, by = NULL, ..., copy = FALSE, in_place
 rows_delete <- function(x, y, by = NULL, ..., copy = FALSE, in_place = FALSE) {
   # Need action = warn, because methods may have
   # side effects that persist even after we abort
-  ellipsis::check_dots_used(action = warn)
+  rlang::check_dots_used(action = warn)
```
Comment (lionel-) [inline R/rows.R:189]: I plan to deprecate the `action = `, maybe wait until that's the case? I'll do it next week.

**Comment by lionel-** [comment]
> Perhaps this should drop the rlang:: prefix too since rlang is @import'ed

yeah for consistency

File: R/rows.R (hunk 1 above)
Comment (lionel-) [reply R/rows.R:189]: Why is this not throwing errors?

Comment (romainfrancois) [reply R/rows.R:189]: cc @krlmlr

**Comment by romainfrancois** [comment]
Not sure why still getting 

```
W  checking dependencies in R code (1.3s)
   '::' or ':::' import not declared from: ‘ellipsis’
```

**Comment by DavisVaughan** [comment]
Is it because the set-ops functions still use `ellipsis::` since that is in a different PR?

**Comment by DavisVaughan** [comment]
Ooooh @romainfrancois this was a tough one. It is because we inline `tidyselect::peek_vars` here and that uses `ellipsis::`
https://github.com/tidyverse/dplyr/blob/04454209ea069939d3335c43846c85c725547a89/R/select-helpers.R#L53

Found by stepping through `tools:::.check_packages_used("dplyr")`

**Comment by romainfrancois** [comment]
This seems to do the trick. Thanks @DavisVaughan. cc @lionel- in case `peek_vars <- tidyselect::peek_vars ` is still needed for something, I don't believe so. 

Perhaps we can use `check_dots_empty()` on the `rows*.data.frame` methods and let implementations (in `dm::` ?) do something different with the `...` ? cc @krlmlr

**Comment by romainfrancois** [comment]
We have: 

```r
rows_insert <- function(x, y, by = NULL, ..., copy = FALSE, in_place = FALSE) {
  # Need action = warn, because methods may have
  # side effects that persist even after we abort
  check_dots_used()
  UseMethod("rows_insert")
}
```

and then in `dm::` 

```r
rows_insert.tbl_dbi <- function(x, y, by = NULL, ...,
                                in_place = NULL, copy = FALSE, check = NULL,
                                returning = NULL) {
```

And the `...` are not passed through, so this is just about checking that specific arguments of the methods (`check=, returning=` are fully named ? Should `dm` rather itself call `check_dots_empty()` so that it fails when that happens rather  warn ?

File: R/rows.R (hunk 1 above)
Comment (krlmlr) [reply R/rows.R]: These methods are unusual in the sense that their main purpose may be the side effects they cause, especially with `in_place = TRUE` for database backends. Throwing an error after the side effects happened would be wrong here.

**Comment by krlmlr** [comment]
Am I understanding this right: if all methods use `check_dots_empty()`, we don't need `check_dots_used()` in the generic?

File: R/rows.R (hunk 1 above)
Comment (DavisVaughan) [reply R/rows.R]: I'm generally in favor of removing the `check_dots_*()` helpers from the generics, and letting the methods call `check_dots_empty()`, like what @romainfrancois did in dm https://github.com/cynkra/dm/pull/639

The only other place we use `check_dots_used()` like this is in `pull()`, but we could change that too to make `pull.data.frame()` call `check_dots_empty()`.

**Comment by romainfrancois** [comment]
Yeah I guess so, at least in this case where IIUC the generic had `check_dots_used()` to make sure everything in the `...` was used after the end of the methods, which as the comment implied might be too late, happen after some changes have been performed in the database, ....

If instead, the method checks that the `...` are unused, it can fail immediately, and so avoid doing unwanted work on the database.

**Review by lionel- — APPROVED** [review]
Another potential reason for using `check_dots_empty()` in dplyr methods instead of `check_dots_used()` in dplyr generics is that the latter can't detect that an argument was defused with `{{` or `enquo()`.

**Pull request merged**